api/app/main.py

```diff
@@ -418,90 +394,37 @@ async def get_analytics_result(resource_id: str):
-        if len(gadm_ids) > 2:
-            select_clause += ", subregion"
-            where_clause += f" AND subregion = '{gadm_ids[2]}'"
-            group_by_clause += ", subregion"
-
-        # Includes an intersection, so group by the appropriate column
-        if intersections:
-            select_clause += f", {intersection_col}"
-            group_by_clause += f", {intersection_col}"
-
-        group_by_clause += ", alert_date, alert_confidence"
-        
-        # Query and make sure output names match the expected schema (?)
-        select_clause += ", alert_date, alert_confidence as confidence, sum(count) as value"
-        query = f"{select_clause} {from_clause} {where_clause} {group_by_clause}"
-        
+        query = create_gadm_dist_query(gadm_id, intersections)
+
         # Dumbly doing this per request since the STS token expires eventually otherwise
-        # According to this issue, duckdb should auto refresh the token in 1.3.0, 
+        # According to this issue, duckdb should auto refresh the token in 1.3.0,
         # but it doesn't seem to work for us and people are reporting the same on the issue
         # https://github.com/duckdb/duckdb-aws/issues/26
         # TODO do this on lifecycle start once autorefresh works
-        duckdb.query('''
+        duckdb.query(
+            """
             CREATE OR REPLACE SECRET secret (
                 TYPE s3,
                 PROVIDER credential_chain
             );
-        ''')
+        """
+        )
 
         # Send query to DuckDB and convert to return format
         alerts_df = duckdb.query(query).df()
     else:
-        if aoi["type"] == "geojson":
-            geojson = aoi["geojson"]
-        else:
-            if aoi["type"] == "key_biodiversity_area":
-                url = f"https://data-api.globalforestwatch.org/dataset/birdlife_key_biodiversity_areas/latest/query?sql=select gfw_geojson from data where sitrecid = {aoi['id']}&x-api-key={API_KEY}"
-            elif aoi["type"] == "protected_area":
-                url = f"https://data-api.globalforestwatch.org/dataset/wdpa_protected_areas/latest/query?sql=select gfw_geojson from data where wdpaid = {aoi['id']}&x-api-key={API_KEY}"
-            elif aoi["type"] == "indigenous_land":
-                url = f"https://data-api.globalforestwatch.org/dataset/landmark_icls/latest/query?sql=select gfw_geojson from data where objectid = {aoi['id']}&x-api-key={API_KEY}"
-
-            response = requests.get(url)
-            response = response.json()
-            if response["data"]:
-                geojson = json.loads(response["data"][0]["gfw_geojson"])
+        geojson = await get_geojson(aoi)
 
         if metadata_content["intersections"]:
             intersection = metadata_content["intersections"][0]
         else:
             intersection = None
 
-        alerts_df = await _analyze(geojson, aoi, intersection)
-        
+        alerts_df = await zonal_statistics(geojson, aoi, intersection)
```

Comment:
We're gonna move to doing this async and right after the `post` aren't we?

Reply:
I might quickly do this as the final refactor for this PR